File: Jan_Basic_Programming/Assignments/C/12.c
#include<stdio.h>
// Decimal to Binary
int main(){
    int decimal;
    int binary[50];

    printf("Enter the decimal Number : ");
    scanf("%d", &decimal);
    
    int i=0;
    while(decimal>0){
        binary[i++]=decimal % 2;
        decimal/=2;
    }

    for(int j=i-1; j>=0; j--){
        printf("%d ", binary[j]);
    }
 return 0;
}

File: Mar_PHP/Assignments/BasicPrograms/12.php
<!DOCTYPE html>
<html>
<head>
    <title>Decimal to Binary Conversion</title>
</head>
<body>

<form method="post">
    <label for="number">Enter a decimal number to convert to binary:</label>
    <input type="number" id="number" name="number">
    <button type="submit">Convert</button>
</form>

<?php
if ($_SERVER["REQUEST_METHOD"] == "POST") {
    $n = $_POST['number'];
    $binaryNum = array();

    $i = 0;
    while ($n > 0) {
        $binaryNum[$i] = $n % 2;
        $n = (int)($n / 2);
        $i++;
    }

    echo "Binary representation: ";
    for ($j = $i - 1; $j >= 0; $j--) {
        echo $binaryNum[$j];
    }
}
?>

</body>
</html>
